scripts/2022_xx_xx_1_arb_claim_and_bridge_admin_fees_in_USDC.py

```diff
@@ -0,0 +1,302 @@
+                swap_address = swap_to_deposit_dict[token_to_swap_dict[token_address]]
+                # Metaswap deposit for swapping
+                swap = Contract.from_abi(
+                    "MetaSwapDeposit", swap_address, SWAP_ABI
+                )
+                # get (flattened) token indices from underlying swap contract
+                meta_swap = Contract.from_abi(
+                    "MetaSwap", token_to_swap_dict[token_address], META_SWAP_ABI
+                )
+                base_swap = Contract.from_abi(
+                    "BaseSwap", meta_swap.metaSwapStorage()[0], SWAP_ABI
+                )
+                base_token_index_to = base_swap.getTokenIndex(USDC_ARBITRUM)
+                token_index_from = 0  # index 0 is non-base-pool token
+                # offset by one for flattened 'to' token index
+                token_index_to = 1 + base_token_index_to
+
+            # deadline 10 mins from now
+            deadline = chain[chain.height].timestamp + 10 * 60
+
+            # min amount to receive
+            min_amount = swap.calculateSwap(
+                token_index_from,
+                token_index_to,
+                amount_to_swap
+            )
+
+            # approve amount to swap
+            token_contract = Contract.from_abi(
+                "ERC20", token_address, ERC20_ABI
+            )
+            print(
+                f"Approving for {token_contract.symbol()} {LP_balance}"
+            )
+            token_contract.approve(
+                swap_address,
+                amount_to_swap,
+                {"from": multisig.address}
+            )
+
+            # perform swap
+            print(
+                f"Swapping {amount_to_swap / (10 ** token_contract.decimals())} {token_contract.symbol()} to USDC"
+            )
+            swap.swap(
+                token_index_from,
+                token_index_to,
+                amount_to_swap,
+                min_amount,
+                deadline,
+                {"from": multisig.address}
+            )
+
+    # bridging USDC to mainnet
+    USDC = Contract.from_abi("ERC20", USDC_ARBITRUM, ERC20_ABI)
+
+    # bridge difference between balance before and after claiming + converting
+    amount_to_bridge = USDC.balanceOf(
+        multisig.address
+    ) - token_balances_before[USDC_ARBITRUM]
+
+    print(
+        f"Bridging ${USDC.symbol()} {amount_to_bridge / (10 ** USDC.decimals())} to mainnet"
+    )
+
+    # find gateway for USDC
+    token_gateway_address = gateway_router.getGateway(
+        USDC.address
+    )
+
+    # approve gateway
+    USDC.approve(
+        token_gateway_address,
+        amount_to_bridge,
+        {"from": MULTISIG_ADDRESSES[CHAIN_IDS[TARGET_NETWORK]]}
+    )
+
+    # send tx to bridge
+    gateway_router.outboundTransfer(
+        USDC_MAINNET,
+        MULTISIG_ADDRESSES[CHAIN_IDS["MAINNET"]],
+        amount_to_bridge,
+        "0x"  # TODO: clarify what format this needs to have
+    )
+
```

Comment:
Correct me if im wrong but this should be written for bringing USDC from arbitrum to mainnet, not the other way around.

Reply:
Indeed. You mean because of line 286? I think they require the token address of the corresponding L1 token. See [here](https://github.com/OffchainLabs/token-bridge-contracts/blob/e304fa491f0574c984e80705f9f6d7cf8f0d798e/contracts/tokenbridge/arbitrum/gateway/L2GatewayRouter.sol#L57)